cmd/example-hook-sidecar/smbios.go

```diff
@@ -0,0 +1,114 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"encoding/xml"
+	"net"
+	"os"
+
+	"google.golang.org/grpc"
+
+	vmSchema "kubevirt.io/kubevirt/pkg/api/v1"
+	hooks "kubevirt.io/kubevirt/pkg/hooks"
+	hooksInfo "kubevirt.io/kubevirt/pkg/hooks/info"
+	hooksV1alpha1 "kubevirt.io/kubevirt/pkg/hooks/v1alpha1"
+	"kubevirt.io/kubevirt/pkg/log"
+	domainSchema "kubevirt.io/kubevirt/pkg/virt-launcher/virtwrap/api"
+)
+
+const baseBoardManufacturerAnnotation = "smbios.vm.kubevirt.io/baseBoardManufacturer"
+
+type infoServer struct{}
+
+func (s infoServer) Info(ctx context.Context, params *hooksInfo.InfoParams) (*hooksInfo.InfoResult, error) {
+	log.Log.Info("Hook's Info method has been called")
+
+	return &hooksInfo.InfoResult{
```

Comment:
ideally the user shouldn't have to know about this. It would be great if the user just declared what version of the api they were using and this info callback was managed internally by us somehow.

Reply:
I do this, so user can implement multiple versions and we have an option to pick the latest supported one. I agree that it's a little too complicated, but when I tried to think about something simpler, it always broke on compatibility issues.

Reply:
I will not be around June 4-16, I'd rather finish this simple form first and add a framework later.

Reply:
> I will not be around June 4-16, I'd rather finish this simple form first and add a framework later.

yup, I agree. Lets get what you've done so far merged first. This can be the next iteration we follow up with.